docs/readmes/proposals/p010_vendor_neutral_dp.md

```diff
@@ -416,7 +456,250 @@ scope of work.
+well defined information about CBSD we suggest to reuse an already existing MySQL / PostgreSQL database
+from the Magma Orchestrator component (with a new schema).
+At the very beginning the database usage would be limited to:
+* storing records for CBSDs configuration (name, location, radio parameters, etc.)
+* keeping track of grants limiting redundant CBSD - SAS connections
+* storing information about operational mode - Active or Passive - per CBSD/DP
+* user administration & access rights within DP system
+
+### Queuing mechanisms 
+
+One of the goals of the Domain Proxy is to provide a stable exchange of information
+between CBSDs and SAS and minimize the network traffic between them.
+Using popular existing queuing mechanisms gives us batching functionality
+that allows us to send less requests to the SAS server.
+
+While it might be partially achieved without a queue,
+by implementing a buffer for batching requests sent from multiple radios,
+the queue has the advantage of stability over a simple buffer.
+
+If Configuration Controller pulls requests from the queue and then fails,
+the requests are gone and the behavior of the system cannot be predicted at this point
+(will the CBSDs timeout and resend their requests?).
+The queue allows for the acknowledgement procedure, where,
+if a message is not processed and acknowledged by the consumer,
+it automatically gets re-queued.
+
+System includes interaction with external components
+that shouldn’t be forced to implement retrying mechanisms
+should there be any connection faults.
+That’s why it is convenient and beneficial to base our input processing on a queue
+which can persist sent requests in the negative scenario.
+
+### Message Queue Pros/Cons
+
+* **RabbitMQ**
+  * Pros:
+    - Out-of-the-box fulfills our requirement for batching requests
+    - Microservice approach: easier to split into components, mock & test
+    - Event-based flow is a reasonable way when there can be multiple consumers
+    - Utilizing a full-blown queue makes the architecture extendable - we don’t need to incorporate a pub/sub solution in each new subscribing component
+    - The component is purely internal - it won’t affect other architecture components built by the community
+
+  * Cons:
+    - Additional infrastructure component to maintain (internal to th DP)
+    - Pub/Sub overhead (minor in our scenario)
+
+* **ZeroMQ**
+  * Pros:
+    - ZeroMQ implements queuing at a higher abstraction level - no need to deal with networking, sockets, etc.
+    - Component is already used in the ecosystem
+
+  * Cons:
+    - It’s a very simple queue, the required functionalities like message batching, persistency, still would need to be implemented and later maintained as separate components (potentially bug-prone)
+    - It would need to be separately added to each component which would need queuing functionalities
+    - Cannot be easily replaced, mocked, up- or downscaled basing on the needs
+
+
```

Comment:
Might be useful to also add a section on metrics we want to keep track of and possible alerts we might want to add, when we integrate this with prometheus / alert manager in the future.

Reply:
@karthiksubraveti 
Good suggestion. Probably at this point in time it might be difficult to define exact telemetry data, but surely some very basics like default process resource usage or number of processed requests, error counters we could list already.

We will try to come up with an intermediate list for the metrics and possibly alert definitions.